app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/FileListAdapter.java

```diff
@@ -0,0 +1,83 @@
+package org.kiwix.kiwixmobile.zim_manager.local_file_transfer;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import org.kiwix.kiwixmobile.R;
+
+import java.util.ArrayList;
+
+/**
+ * Helper class, part of the local file sharing module.
+ *
+ * Defines the Adapter for the list of file-items displayed in {@link TransferProgressFragment}
+ * */
+public class FileListAdapter extends RecyclerView.Adapter<FileListAdapter.FileViewHolder> {
+  private final ArrayList<FileItem> fileItems;
+  private LayoutInflater layoutInflater;
+
+  public FileListAdapter(Context context, ArrayList<FileItem> fileItems) {
+    this.layoutInflater = LayoutInflater.from(context);
+    this.fileItems = fileItems;
+  }
+
+  @NonNull
+  @Override
+  public FileListAdapter.FileViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+    View itemView = layoutInflater.inflate(R.layout.item_transfer_list, parent, false);
```

Comment:
parent is a VIewGroup -so also a view- and has a context so I would not pass in a context to this class, just create layoutInflater here

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/52f8ae577128c09ccff947d51f920eb909375177)